### DISCUSSION TITLE
Use custom fetch(er) for web client

### DISCUSSION THREAD

**Discussion by arfath-linklet** [ClickHouse/clickhouse-js#314] [Ideas]
By default the web client uses the global fetch. The idea is to configure the client with a custom fetch(er) that may decorate the fetch function with additional functionality. One example we encountered -

- Using mTLS: Cloudflare Workers can use mTLS with added bindings, the binding exposes a fetch(er) with mTLS pre-configured https://developers.cloudflare.com/workers/runtime-apis/bindings/mtls/

**Comment by slvrtrn** [comment]
Can you please create an issue for that?

**Reply by arfath-linklet** [reply]
https://github.com/ClickHouse/clickhouse-js/issues/315
